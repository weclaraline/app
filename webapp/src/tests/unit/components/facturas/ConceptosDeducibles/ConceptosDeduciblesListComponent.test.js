import "../../../testing_utils";
import { shallow } from "enzyme";
import ConceptoDeduciblesListComponent from "../../../../../components/facturas/ConceptosDeducibles/ConceptosDeduciblesListComponent";
import { generateDeductibleTypesAndRecommendations } from "../generation_service";
import toJson from "enzyme-to-json";
import { render, unmountComponentAtNode } from "react-dom";
import { getByTestId, waitFor } from "@testing-library/react";

let container = null;

const build = (props) => render(<ConceptoDeduciblesListComponent conceptosDeducibles={props}/>, container);

const conceptoDeducibleProps =  generateDeductibleTypesAndRecommendations(4);

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});
  
afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
    return container;
});

it("Renders original content", () => {
    const tree = shallow(<ConceptoDeduciblesListComponent conceptosDeducibles={[]}/>);
    expect(toJson(tree)).toMatchSnapshot();
});

it("Renders elements passed by props", async () => {
    // const tree = shallow(<ConceptoDeduciblesListComponent conceptosDeducibles={conceptoDeducibleProps}/>);
    build(conceptoDeducibleProps);
    await waitFor(()=> {
        conceptoDeducibleProps.forEach( ({key}) => {
            expect(getByTestId(container, `concepto_deducible_${key}`)).toBeInTheDocument();
        });
    });
});