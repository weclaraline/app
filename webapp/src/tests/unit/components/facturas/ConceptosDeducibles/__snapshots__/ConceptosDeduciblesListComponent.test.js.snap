// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<Styled(MuiBox)
  mb={2}
>
  <WithStyles(ForwardRef(Typography))
    align="left"
    gutterBottom={true}
    variant="h6"
  >
    No se encontraron resultados
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;
